src/main/java/teammates/ui/controller/FeedbackParticipantFeedbackResponseCommentDeleteAction.java

```diff
@@ -0,0 +1,90 @@
+import teammates.common.datatransfer.attributes.StudentAttributes;
+import teammates.common.util.Assumption;
+import teammates.common.util.Const;
+import teammates.ui.pagedata.FeedbackResponseCommentAjaxPageData;
+
+public class FeedbackParticipantFeedbackResponseCommentDeleteAction extends Action {
+
+    protected String courseId;
+    protected String feedbackSessionName;
+    protected boolean isModeration;
+
+    @Override
+    protected ActionResult execute() {
+        courseId = getRequestParamValue(Const.ParamsNames.COURSE_ID);
+        Assumption.assertPostParamNotNull(Const.ParamsNames.COURSE_ID, courseId);
+        feedbackSessionName = getRequestParamValue(Const.ParamsNames.FEEDBACK_SESSION_NAME);
+        Assumption.assertPostParamNotNull(Const.ParamsNames.FEEDBACK_SESSION_NAME, feedbackSessionName);
+        String feedbackResponseId = getRequestParamValue(Const.ParamsNames.FEEDBACK_RESPONSE_ID);
+        Assumption.assertPostParamNotNull(Const.ParamsNames.FEEDBACK_RESPONSE_ID, feedbackResponseId);
+        String feedbackResponseCommentId = getRequestParamValue(Const.ParamsNames.FEEDBACK_RESPONSE_COMMENT_ID);
+        Assumption.assertPostParamNotNull(Const.ParamsNames.FEEDBACK_RESPONSE_COMMENT_ID, feedbackResponseCommentId);
+        FeedbackSessionAttributes session = logic.getFeedbackSession(feedbackSessionName, courseId);
+        FeedbackResponseAttributes response = logic.getFeedbackResponse(feedbackResponseId);
+        Assumption.assertNotNull(response);
+
+        isModeration = false;
+        if (getRequestParamValue(Const.ParamsNames.FEEDBACK_SESSION_MODERATED_PERSON) != null) {
+            isModeration = true;
+        }
+
+        Long commentId = Long.parseLong(feedbackResponseCommentId);
+
+        verifyAccessibleForUserToFeedbackResponseComment(session, response, commentId);
+
+        logic.deleteDocumentByCommentId(commentId);
+        logic.deleteFeedbackResponseCommentById(commentId);
+
+        appendToStatusToAdmin(commentId);
+
+        FeedbackResponseCommentAjaxPageData data =
+                new FeedbackResponseCommentAjaxPageData(account, sessionToken);
+
+        return createAjaxResult(data);
+    }
+
+    private void verifyAccessibleForUserToFeedbackResponseComment(FeedbackSessionAttributes session,
+            FeedbackResponseAttributes response, Long commentId) {
+        if (isModeration) {
+            InstructorAttributes instructor = logic.getInstructorForGoogleId(courseId, account.googleId);
+            gateKeeper.verifyAccessible(instructor, session, false, response.giverSection,
+                    Const.ParamsNames.INSTRUCTOR_PERMISSION_SUBMIT_SESSION_IN_SECTIONS);
+            gateKeeper.verifyAccessible(instructor, session, false, response.recipientSection,
+                    Const.ParamsNames.INSTRUCTOR_PERMISSION_SUBMIT_SESSION_IN_SECTIONS);
+        } else {
+            FeedbackResponseCommentAttributes frc = logic.getFeedbackResponseComment(commentId);
```

Comment:
You still need to check `isCommentFromFeedbackParticipant` here

Reply:
I am checking it. Isn't that sufficient?
```
    public void verifyAccessibleForDeletion(FeedbackResponseCommentAttributes frc, String feedbackParticipant) {
        verifyNotNull(frc, "feedback response comment");
        verifyNotNull(frc.commentGiver, "feedback response comment giver");
        verifyNotNull(feedbackParticipant, "comment giver");

        if (!frc.isCommentFromFeedbackParticipant || !frc.commentGiver.equals(feedbackParticipant)) {
            throw new UnauthorizedAccessException("Comment [" + frc.getId() + "] is not accessible to " + feedbackParticipant);
        }
    }
```

Reply:
It is better to separate the logic for isCommentFromFeedbackParticipant here.

In addition
`verifyAccessibleForUserToFeedbackResponseComment` -> `verifyDeletePermissionForUserToFeedbackResponseComment `